nuxeo-features/nuxeo-platform-oauth/src/main/java/org/nuxeo/ecm/platform/oauth2/providers/OAuth2ServiceProviderRegistryImpl.java

```diff
@@ -69,30 +69,57 @@ public OAuth2ServiceProvider getProvider(String serviceName) {
             filter.put("serviceName", serviceName);
 
             List<DocumentModel> providers = queryProviders(filter, 1);
-            return providers.isEmpty() ? null : buildProvider(providers.get(0));
+            return providers.isEmpty() ? null : providers.get(0);
         } catch (DirectoryException e) {
             log.error("Unable to read provider from Directory backend", e);
             return null;
         }
     }
 
     @Override
+    public OAuth2ServiceProvider getProvider(String serviceName) {
+        DocumentModel model = getProviderDocModel(serviceName);
+        return model != null ? buildProvider(model): null;
```

Comment:
I prefer `model == null ? null : buildProvider(model)` to avoid negations. Also please make sure of having a space left and right of the `:`.